packages/slonik/src/factories/createQueryValidator.ts

```diff
@@ -0,0 +1,41 @@
+import { type Query } from '../types';
+import { InputSyntaxError, InvalidInputError } from '@slonik/errors';
+import { stripSlonikPlaceholders } from '@slonik/sql-tag';
+import { parseAsync } from 'pgsql-parser';
+
+export const createQueryValidator = () => {
+  const queries: Record<string, number> = {};
+
+  return async (query: Query) => {
+    let stats = queries[query.sql];
+
+    if (stats === undefined) {
+      const ast = await parseAsync(stripSlonikPlaceholders(query.sql));
```

Comment:
@gajus you're calling this twice -- once here and again inside `executeQuery` before this is called

Reply:
Refactored